doc/source/reference/c-api/types-and-structures.rst

```diff
@@ -509,6 +512,17 @@ PyArrayDescr_Type and PyArray_Descr
     functions can (and must) deal with mis-behaved arrays. The other
     functions require behaved memory segments.
 
+    .. note::
+        The functions are largely legacy API, however, some are still used.
+        As of NumPy 2.x they are only available via `PyDataType_GetArrFuncs`
+        (see the function for more details).
+        Before using any function defined in the struct you should check
+        whether it is ``NULL``.  In general, the functions ``getitem``,
+        ``setitem``, ``copyswap``, and ``copyswapn`` can be expected to be
+        defined, but all functions are expected to be replaced with newer API.
+        For example, ``PyArray_Pack`` is a more powerful version of ``setitem``
```

Comment:
`PyArray_Pack` still isn't in the public API

Reply:
Right, but let's make sure we add it?  Should have done that earlier anyway...

Reply:
I added a note about this to #25820